com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationManager.java

```diff
@@ -234,40 +235,59 @@ public void deleteNotificationChannel(String id) {
     }
 
     // This is called from Unity managed code to call AlarmManager to set a broadcast intent for sending a notification.
-    public void scheduleNotificationIntent(Intent data_intent_source) {
-        // TODO: why we serialize/deserialize again?
-        String temp = UnityNotificationUtilities.serializeNotificationIntent(data_intent_source);
-        Intent data_intent = UnityNotificationUtilities.deserializeNotificationIntent(mContext, temp);
+    public void scheduleNotification(Notification.Builder notificationBuilder) {
+        Bundle extras = notificationBuilder.getExtras();
+        int id = extras.getInt("id", -1);
+        long repeatInterval = extras.getLong("repeatInterval", -1);
+        long fireTime = extras.getLong("fireTime", -1);
+        Notification notification = notificationBuilder.build();
+
+        Intent openAppIntent = UnityNotificationManager.buildOpenAppIntent(mContext, mOpenActivity);
+        openAppIntent.putExtra("unityNotification", notification);
+        PendingIntent pendingIntent = getActivityPendingIntent(mContext, id, openAppIntent, 0);
 
-        int id = data_intent.getIntExtra("id", 0);
+        // if less than a second in the future, notify right away
+        if (fireTime - Calendar.getInstance().getTime().getTime() < 1000) {
+            notificationBuilder.setContentIntent(pendingIntent);
+            finalizeNotificationForDisplay(mContext, notificationBuilder);
+            notification = notificationBuilder.build();
+            notify(mContext, id, notification);
+            if (repeatInterval <= 0)
+                return;
```

Comment:
I'm wondering if we can skip this quick pass if it's repeatable? I mean only notify right away if it's not repeatable.

Reply:
You can't repeat it frequently and it doesn't make sense either, it would be a terrible user experience if notification would pop so often, so I think we can assume here the value makes sense.